ts/src/coinmetro.ts

```diff
@@ -0,0 +1,1922 @@
+        const currencyIds = this.safeValue (this.options, 'currencyIdsListForParseMarket', []);
+        for (let i = 0; i < currencyIds.length; i++) {
+            const currencyId = currencyIds[i];
+            const entryIndex = marketId.indexOf (currencyId);
+            if (entryIndex !== -1) {
+                const restId = marketId.replace (currencyId, '');
+                if (this.inArray (restId, currencyIds)) {
+                    if (entryIndex === 0) {
+                        baseId = currencyId;
+                        quoteId = restId;
+                    } else {
+                        baseId = restId;
+                        quoteId = currencyId;
+                    }
+                    break;
+                }
+            }
+        }
+        const result = {
+            'baseId': baseId,
+            'quoteId': quoteId,
+        };
+        return result;
+    }
+
+    parseMarketPrecisionAndLimits (currencyId) {
+        const currencies = this.safeValue (this.options, 'currenciesByIdForParseMarket', {});
+        const currency = this.safeValue (currencies, currencyId, {});
+        const precision = this.safeInteger (currency, 'precision');
+        const limits = this.safeValue (currency, 'limits', {});
+        const amountLimits = this.safeValue (limits, 'amount', {});
+        const minLimit = this.safeNumber (amountLimits, 'min');
+        const result = {
+            'precision': precision,
+            'minLimit': minLimit,
+        };
+        return result;
+    }
+
+    async fetchOHLCV (symbol: string, timeframe = '1m', since: Int = undefined, limit: Int = undefined, params = {}): Promise<OHLCV[]> {
+        /**
+         * @method
+         * @name coinmetro#fetchOHLCV
+         * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://documenter.getpostman.com/view/3653795/SVfWN6KS#13cfb5bc-7bfb-4847-85e1-e0f35dfb3573
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int} [since] timestamp in ms of the earliest candle to fetch
+         * @param {int} [limit] the maximum amount of candles to fetch
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {int} [params.until] the latest time in ms to fetch entries for
+         * @returns {int[][]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'pair': market['id'],
+            'timeframe': this.safeString (this.timeframes, timeframe, timeframe),
+        };
+        let until = undefined;
+        if (since !== undefined) {
+            if ((since < 0) || (!Number.isInteger (since))) {
```

Comment:
The exchange returns not informative message `"No more than 1000 candles per call"` if `since` is not a non-negative integer. With this check the user will get more information about his mistake. Should I delete this check?

Reply:
Yes you can delete it because it is expected to be an integer by definition